### ISSUE TITLE
So sad this project was discontinued

### ISSUE THREAD

**Issue by carnivash** [adnathan/ace#75]
It saved my life many times, so easy to integrate native code into cordova, does anyone know about a similar project?

**Comment by andrecbr** [comment]
+1

**Comment by baldram** [comment]
@carnivash Yes. There is a similar project called [Tabris.js](https://github.com/eclipsesource/tabris-js) and what is important an active and matured one. They've just released [beta of 2.0 version](http://eclipsesource.com/blogs/2017/02/08/tabris-js-2-0-beta-1-is-here/). As you can learn from [this article](http://eclipsesource.com/blogs/2015/03/02/apache-cordova-vs-tabris-js/), it offers quite similar capabilities to Ace. It uses Cordova's plugin API to provide an access to native UI elements. 

I can't say much more than I read about it, but please share your opinion when you compare both Tabris and Ace. 

I also wonder how about Tabris vs NativeScript or React Native. If you have any thoughts about this, could you please share them?

**Comment by cordovapolymer** [comment]
@baldram looks nice, but unfortunately its native components are closed source.

**Comment by baldram** [comment]
@cordovapolymer I'm not that sure. Let's take a look on this plugin: https://github.com/eclipsesource/tabris-plugin-maps . You have full Android and iOS native source, not only JS examples.
So even if there are no source for the core part, you have at least a control on this and other 3rd party add-ons. Plus possibility to create own ones. So you are able to build your own solution on the top of Tabris.

When considering this technology, we can see that EclipseSource is a foundation with a long history.
There is a growing community, the Github project and Slack support is very active. They have built a business around this solution. Finally it looks like they simply develop it for own projects. I think they are trustworthy if you like this solution.

I must admit I have no interest to write this everything. Those are just my observations I want to share.
The decision is yours ;)

**Comment by uniphonic** [comment]
Why go for an alternative, when you could still use ACE? I think with ACE being open source as it is, anyone could fork it and pick up where Microsoft left off, and continue development with it. Right?

**Comment by cordovapolymer** [comment]
@uniphonic , nobody did it so far https://github.com/Microsoft/ace/network

**Comment by delanick** [comment]
Can someone please pick this up? It's a genius initiative. Tabris.js is a good effort, but I refuse to pay their rates or be limited to one private GIT repo. Stupid commercial models. These companies need to realise that we want the dev tools for free and are prepared to pay for back-end services, like hosting, analytics etc... and even the deployment platform. The IDE and build tools should be free, with a cloud build option, the way Nativescript Sidekick and React Native's Expo does...

**Comment by Textras** [comment]
+1

**Comment by bot101** [comment]
+1 for @delanick

**Comment by DavidGOrtega** [comment]
@delanick, thoughts like yours make that projects like this are never going to be rescued. Develop this framework is a huge effort, so, following your thoughts, if someone take care of they are not making money if they don't work much more creating a building cloud... of course they will have to have a free plan with unlimited projects... all that for you to make money using their work... wow you make me cry and laugh at the same time

**Comment by delanick** [comment]
@DavidGOrtega that's a cute response, spoken like someone only in it for the money (or perhaps a representative of Tabrisjs??). I never said the tools all had to be free, but if it's free with React Native, which is far more complex than Tabrisjs, why can't others do it. I don't mind paying for cloud services, like back-ends, but the IDE should be free. Most, if not all, Cordova plugins are free, which is what this post is about - a Cordova plugin created by Microsoft for native UI, which was FREE, might I add... My entire point was that why pay for Tabris' crazy-expensive plans when others like Microsoft were giving the plugin away for free? It's just a plugin, not an entire environment. I'm not asking for an entire environment like Tabris, I just want someone to fork this specific plugin so I can use it - so the rest of us can use it. If I had the skill I would do it myself and release it for free, as I believe that level of knowledge should be free to all as it further raises the general standard of cross-platform apps that get developed. 

There are many services that offer cloud builds, even Intel XDK offered it for free. They just stopped it as they felt there are many other more mature services which could do it better. But they never had any interest in charging for their technology. I say it again, because you obviously missed it... I don't mind paying for back-end services, never have. The dev tools should be free as that's what the world wants. And again, Tabrisjs is just insanely expensive!! I really doubt their business is making any real money as I don't believe their commercial model is at all sustainable.

**Comment by cordovapolymer** [comment]
@delanick , tabris.js [had a discussion for opening](https://github.com/eclipsesource/tabris-js/issues/1240#issuecomment-308379056) their core, but it seems stalled.

**Comment by delanick** [comment]
@cordovapolymer I would love to have that happen. Pity it's stalled.

**Comment by delanick** [comment]
@cordovapolymer sad part is I've played with Tabris and really enjoy using it, far more than Nativescript even, but small one-man business, especially from South Africa (like me) won't spend that money on those solutions, it's just not feasible. If Tabris was open source and free to use it would be the perfect tool!

**Comment by DavidGOrtega** [comment]
@delanick I'm not associate to Tabris in any way. I'm interested in buying their product. I truly understand that coding wastes many human hours that have to be PAID. If you don't want to pay for someone else work don't blame them just because they don't give it to you for free. I wonder what do you have in your private repos that you don't want others to have a peek and pick YOUR code.

Tools don't have to be nor free nor open source, people is free of doing whatever they want with their work. If you had the knowledge to work in this framework as you state I bet you that you would not maintain it, specially for free, maintaining an open source repo is a huge amount of work. Microsoft, React (facebook), etc... are HUGE teams with HUGE resources, don't spect Tabris or anyone else to be that huge... If you are wondering who is paying them... I will and many others like me I presume.

I wonder why people here is asking Tabris to be open source... This project is OPEN SOURCE and no one is actually specting to work on top of this! You have to use the word FREE as FREE BEER and the real thing is that if Tabris would be FREE as FREE BEER you wont mind if it is closed or open sourced...
